src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -198,18 +219,27 @@ object ScalafixPlugin extends AutoPlugin {
         throw new ScalafixFailed(List(ScalafixError.CommandLineError))
     }
     val rulesDepsExternal = parsed.map(_.dependency)
-    val refreshClasspath =
-      (projectDepsInternal ++ projectDepsExternal ++ rulesDepsExternal).nonEmpty
+    val projectDepsInternal0 = projectDepsInternal.filter {
+      case directory if directory.isDirectory =>
+        directory.**(AllPassFilter).get.exists(_.isFile)
+      case file if file.isFile => true
+      case _ => false
+    }
+    val customToolClasspath =
+      (projectDepsInternal0 ++ projectDepsExternal ++ rulesDepsExternal).nonEmpty
     val interface =
-      if (refreshClasspath)
-        scalafixInterface().withArgs(
-          ToolClasspath(
-            projectDepsInternal.map(_.toURI.toURL),
-            baseDepsExternal ++ projectDepsExternal ++ rulesDepsExternal,
-            baseResolvers
-          )
+      if (customToolClasspath) {
+        val toolClasspath = ToolClasspath(
+          projectDepsInternal0.map(_.toURI.toURL),
```

Comment:
Unrelated to this PR, I see that we're using `java.net.URL` as cache keys. This is problematic because `URL.equals` does remote blocking calls

> Two hosts are considered equivalent if both host names can be resolved into the same IP addresses; else if either host name can't be resolved, the host names must be equal without regard to case; or both host names equal to null.

https://docs.oracle.com/javase/6/docs/api/java/net/URL.html#equals(java.lang.Object)

It's probably better to use `URI` instead of `URL`

Reply:
:+1: already got bitten by that, but apparently not enough to care... will change that in a separate PR